File: LeetCode/moveZeroes.py
# 283

def moveZeroes(nums):
    """
    :type nums: List[int]
    :rtype: void Do not return anything, modify nums in-place instead.
    """

    zero = 0  # records the position of "0"
    for i in range(len(nums)):
        if nums[i] != 0:
            nums[i], nums[zero] = nums[zero], nums[i]
            print(nums)
            zero += 1
            print(zero)

moveZeroes([0,1,0,3,5,13])

File: LeetCode/CSharp/MoveZeroes.cs
public class Solution {
    public void MoveZeroes(int[] nums) {
        int zero = 0;
        for(int i = 0; i < nums.Length; i++) {
            if(nums[i] != 0) {
                int temp = nums[i];
                nums[i] = nums[zero];
                nums[zero] = temp;
                zero += 1;
            }
        }       
    }
}